commit 091415f3998826d77509c9502973d3a74568014c
Author: Powei Feng <no-reply@google.com>
Date:   Mon Apr 20 16:41:20 2026 +0000

    jni api checks

diff --git a/android/filament-android/src/main/java/com/google/android/filament/Material.java b/android/filament-android/src/main/java/com/google/android/filament/Material.java
index f14d1d2c7c..f33053d3c1 100644
--- a/android/filament-android/src/main/java/com/google/android/filament/Material.java
+++ b/android/filament-android/src/main/java/com/google/android/filament/Material.java
@@ -394,11 +394,16 @@ public class Material {
         }
     }
 
+    // LINT.IfChange(api)
+    @UsedByNative("material_java_wrappers.h")
     public Material(long nativeMaterial) {
         mNativeObject = nativeMaterial;
         long nativeDefaultInstance = nGetDefaultInstance(nativeMaterial);
         mDefaultInstance = new MaterialInstance(this, nativeDefaultInstance);
     }
+    // LINT.ThenChange(
+    //     //depot/google3/third_party/arcore/ar/sceneform/loader/model/model_material_jni.cc:api
+    // )
 
     public static class Builder {
         public enum ShadowSamplingQuality {
diff --git a/android/filament-android/src/main/java/com/google/android/filament/MaterialInstance.java b/android/filament-android/src/main/java/com/google/android/filament/MaterialInstance.java
index d386ab279b..4e86b3c8ec 100644
--- a/android/filament-android/src/main/java/com/google/android/filament/MaterialInstance.java
+++ b/android/filament-android/src/main/java/com/google/android/filament/MaterialInstance.java
@@ -16,6 +16,8 @@
 
 package com.google.android.filament;
 
+import com.google.android.filament.proguard.UsedByNative;
+
 import androidx.annotation.IntRange;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
@@ -118,6 +120,16 @@ public class MaterialInstance {
         mNativeMaterial = nGetMaterial(mNativeObject);
     }
 
+    // LINT.IfChange(api)
+    @UsedByNative("material_java_wrappers.h")
+    MaterialInstance(long nativeMaterial, long nativeMaterialInstance) {
+        mNativeMaterial = nativeMaterial;
+        mNativeObject = nativeMaterialInstance;
+    }
+    // LINT.ThenChange(
+    //     //depot/google3/third_party/arcore/ar/sceneform/loader/model/model_material_jni.cc:api
+    // )
+
     /**
      * Creates a new {@link #MaterialInstance} using another {@link #MaterialInstance} as a template for initialization.
      * The new {@link #MaterialInstance} is an instance of the same {@link Material} of the template instance and
