diff --git a/filament/backend/src/opengl/platforms/PlatformCocoaTouchGL.mm b/filament/backend/src/opengl/platforms/PlatformCocoaTouchGL.mm
index 91fb5cb88..8a6c52e9b 100644
--- a/filament/backend/src/opengl/platforms/PlatformCocoaTouchGL.mm
+++ b/filament/backend/src/opengl/platforms/PlatformCocoaTouchGL.mm
@@ -39,7 +39,6 @@ using namespace backend;
 struct PlatformCocoaTouchGLImpl {
     EAGLContext* mGLContext = nullptr;
     CAEAGLLayer* mCurrentGlLayer = nullptr;
-    std::vector<CAEAGLLayer*> mHeadlessGlLayers;
     std::vector<EAGLContext*> mAdditionalContexts;
     CGRect mCurrentGlLayerRect;
     GLuint mDefaultFramebuffer = 0;
@@ -119,28 +118,10 @@ Platform::SwapChain* PlatformCocoaTouchGL::createSwapChain(void* nativewindow, u
 }
 
 Platform::SwapChain* PlatformCocoaTouchGL::createSwapChain(uint32_t width, uint32_t height, uint64_t flags) noexcept {
-    CAEAGLLayer* glLayer = [CAEAGLLayer layer];
-    glLayer.frame = CGRectMake(0, 0, width, height);
-
-    // adding the pointer to the array retains the CAEAGLLayer
-    pImpl->mHeadlessGlLayers.push_back(glLayer);
-  
-    return (__bridge SwapChain*) glLayer;
+    return nullptr;
 }
 
 void PlatformCocoaTouchGL::destroySwapChain(Platform::SwapChain* swapChain) noexcept {
-    CAEAGLLayer* glLayer = (__bridge CAEAGLLayer*) swapChain;
-
-    if (pImpl->mCurrentGlLayer == glLayer) {
-        pImpl->mCurrentGlLayer = nullptr;
-    }
-  
-    auto& v = pImpl->mHeadlessGlLayers;
-    auto it = std::find(v.begin(), v.end(), glLayer);
-    if(it != v.end()) {
-        // removing the pointer from the array releases the CAEAGLLayer
-        v.erase(it);
-    }
 }
 
 uint32_t PlatformCocoaTouchGL::getDefaultFramebufferObject() noexcept {
