diff --git a/libs/utils/include/private/utils/android/Tracing.h b/libs/utils/include/private/utils/android/Tracing.h
index 3b25edc19..9ad8a8bf3 100644
--- a/libs/utils/include/private/utils/android/Tracing.h
+++ b/libs/utils/include/private/utils/android/Tracing.h
@@ -19,9 +19,10 @@
 
 #include <stdint.h>
 
 #ifdef FILAMENT_ENABLE_PERFETTO
 
-#include <perfetto/perfetto.h>
+#include "third_party/perfetto/include/perfetto/tracing/track_event.h"
+#include "third_party/perfetto/include/perfetto/tracing/track_event_category_registry.h"
 
 PERFETTO_DEFINE_CATEGORIES_IN_NAMESPACE(tracing,
         perfetto::Category(FILAMENT_TRACING_CATEGORY_FILAMENT),
diff --git a/libs/utils/src/android/Tracing.cpp b/libs/utils/src/android/Tracing.cpp
index 40d54113d..646dc8b49 100644
--- a/libs/utils/src/android/Tracing.cpp
+++ b/libs/utils/src/android/Tracing.cpp
@@ -17,7 +17,9 @@
 #include "filament/libs/utils/include/utils/compiler.h"
 #include <private/utils/Tracing.h>
 
-#include <perfetto/perfetto.h>
+#include "third_party/perfetto/include/perfetto/tracing/backend_type.h"
+#include "third_party/perfetto/include/perfetto/tracing/tracing.h"
+#include "third_party/perfetto/include/perfetto/tracing/track_event.h"
 
 PERFETTO_TRACK_EVENT_STATIC_STORAGE_IN_NAMESPACE(tracing);

