diff --git a/absl/base/BUILD.bazel b/absl/base/BUILD.bazel
index 2b380e61..46c7fd7a 100644
--- a/absl/base/BUILD.bazel
+++ b/absl/base/BUILD.bazel
@@ -211,7 +211,7 @@ cc_library(
         "@rules_cc//cc/compiler:msvc-cl": [],
         "@rules_cc//cc/compiler:clang-cl": [],
         "@rules_cc//cc/compiler:emscripten": [],
-        "//conditions:default": ["-pthread"],
+        "//conditions:default": [],
     }) + ABSL_DEFAULT_LINKOPTS,
     visibility = [
         "//visibility:public",
@@ -278,7 +278,7 @@ cc_library(
             "-ladvapi32",
         ],
         "@rules_cc//cc/compiler:emscripten": [],
-        "//conditions:default": ["-pthread"],
+        "//conditions:default": [],
     }) + ABSL_DEFAULT_LINKOPTS,
     deps = [
         ":atomic_hook",
diff --git a/absl/log/internal/BUILD.bazel b/absl/log/internal/BUILD.bazel
index 005861f9..f68f7113 100644
--- a/absl/log/internal/BUILD.bazel
+++ b/absl/log/internal/BUILD.bazel
@@ -112,6 +112,7 @@ cc_library(
         "//absl/base:config",
         "//absl/base:core_headers",
     ],
+    alwayslink = True,
 )
 
 cc_library(
diff --git a/absl/synchronization/BUILD.bazel b/absl/synchronization/BUILD.bazel
index 9a1aa833..a15d98da 100644
--- a/absl/synchronization/BUILD.bazel
+++ b/absl/synchronization/BUILD.bazel
@@ -130,7 +130,7 @@ cc_library(
         "@rules_cc//cc/compiler:msvc-cl": [],
         "@rules_cc//cc/compiler:clang-cl": [],
         "@rules_cc//cc/compiler:emscripten": [],
-        "//conditions:default": ["-pthread"],
+        "//conditions:default": [],
     }) + ABSL_DEFAULT_LINKOPTS,
     visibility = ["//visibility:public"],
     deps = [
