diff --git a/build_defs/cpp_opts.bzl b/build_defs/cpp_opts.bzl
index c0116bd76..525954dd3 100644
--- a/build_defs/cpp_opts.bzl
+++ b/build_defs/cpp_opts.bzl
@@ -36,17 +36,17 @@ LINK_OPTS = select({
         "Shell32.lib",
     ],
     "@platforms//os:macos": [
-        "-lpthread",
+        "-pthread",
         "-lm",
         "-framework CoreFoundation",
     ],
     "@platforms//os:windows": [
         "-ldbghelp",
-        "-lpthread",
+        "-pthread",
         "-lm",
     ],
     "//conditions:default": [
-        "-lpthread",
+        "-pthread",
         "-lm",
     ],
 })
