diff --git a/include/SDL_config.h b/include/SDL_config.h
index 7e0340cdf..3927701e9 100644
--- a/include/SDL_config.h
+++ b/include/SDL_config.h
@@ -41,6 +41,8 @@
 #include "SDL_config_android.h"
 #elif defined(__PSP__)
 #include "SDL_config_psp.h"
+#elif defined(__LINUX__)
+#include "SDL_config_linux.h"
 #else
 /* This is a minimal configuration just to get SDL running on new platforms */
 #include "SDL_config_minimal.h"
