diff --git a/BUILD.bazel b/BUILD.bazel
index 35dfd665..914619af 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -82,7 +82,7 @@ genrule(
     outs = ["generators.inc"],
     cmd = "$(location generate_registry_tables) --xml=$(location @spirv_headers//:spirv_xml_registry) --generator-output=$(location generators.inc)",
     cmd_bat = "$(location //:generate_registry_tables) --xml=$(location @spirv_headers//:spirv_xml_registry) --generator-output=$(location generators.inc)",
-    exec_tools = [":generate_registry_tables"],
+    tools = [":generate_registry_tables"],
 )
 
 py_binary(
@@ -96,7 +96,7 @@ genrule(
     outs = ["build-version.inc"],
     cmd = "SOURCE_DATE_EPOCH=0 $(location update_build_version) $(location CHANGES) $(location build-version.inc)",
     cmd_bat = "set SOURCE_DATE_EPOCH=0  && $(location //:update_build_version) $(location CHANGES) $(location build-version.inc)",
-    exec_tools = [":update_build_version"],
+    tools = [":update_build_version"],
 )
 
 # Libraries
diff --git a/build_defs.bzl b/build_defs.bzl
index 71891372..dae86797 100644
--- a/build_defs.bzl
+++ b/build_defs.bzl
@@ -1,7 +1,7 @@
 COMMON_COPTS = [
-        "-DSPIRV_CHECK_CONTEXT",
-        "-DSPIRV_COLOR_TERMINAL",
-    ] + select({
+    "-DSPIRV_CHECK_CONTEXT",
+    "-DSPIRV_COLOR_TERMINAL",
+] + select({
     "@bazel_tools//src/conditions:windows": [""],
     "//conditions:default": [
         "-DSPIRV_LINUX",
@@ -76,7 +76,7 @@ def generate_core_tables(version = None):
             "--core-insts-output=$(location {3}) " +
             "--operand-kinds-output=$(location {4})"
         ).format(*fmtargs),
-        exec_tools = [":generate_grammar_tables"],
+        tools = [":generate_grammar_tables"],
         visibility = ["//visibility:private"],
     )
 
@@ -113,7 +113,7 @@ def generate_enum_string_mapping(version = None):
             "--extension-enum-output=$(location {3}) " +
             "--enum-string-mapping-output=$(location {4})"
         ).format(*fmtargs),
-        exec_tools = [":generate_grammar_tables"],
+        tools = [":generate_grammar_tables"],
         visibility = ["//visibility:private"],
     )
 
@@ -139,7 +139,7 @@ def generate_opencl_tables(version = None):
             "--extinst-opencl-grammar=$(location {0}) " +
             "--opencl-insts-output=$(location {1})"
         ).format(*fmtargs),
-        exec_tools = [":generate_grammar_tables"],
+        tools = [":generate_grammar_tables"],
         visibility = ["//visibility:private"],
     )
 
@@ -165,7 +165,7 @@ def generate_glsl_tables(version = None):
             "--extinst-glsl-grammar=$(location {0}) " +
             "--glsl-insts-output=$(location {1})"
         ).format(*fmtargs),
-        exec_tools = [":generate_grammar_tables"],
+        tools = [":generate_grammar_tables"],
         visibility = ["//visibility:private"],
     )
 
@@ -193,7 +193,7 @@ def generate_vendor_tables(extension, operand_kind_prefix = ""):
             "--vendor-insts-output=$(location {1}) " +
             "--vendor-operand-kind-prefix={2}"
         ).format(*fmtargs),
-        exec_tools = [":generate_grammar_tables"],
+        tools = [":generate_grammar_tables"],
         visibility = ["//visibility:private"],
     )
 
@@ -216,7 +216,7 @@ def generate_extinst_lang_headers(name, grammar = None):
             "--extinst-grammar=$< " +
             "--extinst-output-path=$(location {0})"
         ).format(*fmtargs),
-        exec_tools = [":generate_language_headers"],
+        tools = [":generate_language_headers"],
         visibility = ["//visibility:private"],
     )
 
diff --git a/source/opt/convert_to_sampled_image_pass.cpp b/source/opt/convert_to_sampled_image_pass.cpp
index e84d3578..323581ef 100644
--- a/source/opt/convert_to_sampled_image_pass.cpp
+++ b/source/opt/convert_to_sampled_image_pass.cpp
@@ -222,16 +222,26 @@ void ConvertToSampledImagePass::FindUsesOfImage(
   def_use_mgr->ForEachUser(image, [uses, this](Instruction* user) {
     switch (user->opcode()) {
       case SpvOpImageFetch:
+        [[fallthrough]];
       case SpvOpImageRead:
+        [[fallthrough]];
       case SpvOpImageWrite:
+        [[fallthrough]];
       case SpvOpImageQueryFormat:
+        [[fallthrough]];
       case SpvOpImageQueryOrder:
+        [[fallthrough]];
       case SpvOpImageQuerySizeLod:
+        [[fallthrough]];
       case SpvOpImageQuerySize:
+        [[fallthrough]];
       case SpvOpImageQueryLevels:
+        [[fallthrough]];
       case SpvOpImageQuerySamples:
+        [[fallthrough]];
       case SpvOpImageSparseFetch:
         uses->push_back(user);
+        [[fallthrough]];
       default:
         break;
     }
